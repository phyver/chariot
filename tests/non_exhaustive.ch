:set continue_on_error
:set dont_show_priorities

data nat where Zero : nat | Succ : nat -> nat

(* INCOMPLETE *)
val f 0 = ???
  | f (n+2) = ???


data list('x) where
    | Nil : list('x)
    | Cons : 'x -> list('x) -> list('x)

:set dont_check_completeness

val last [x] = x
  | last (x::xs) = last xs

:show last

:unset dont_check_completeness

val last2 [x] = x
  | last2 (x::xs) = last2 xs


