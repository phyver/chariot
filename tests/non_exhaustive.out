[31m--** [0mfunction f is not complete
val
   last : list('a) â†’ 'a
    | last [x] = x
    | last (x::xs) = last xs

[31m--** [0mfunction last2 is not complete
